Rubberduck.Parsing/VBA/ParseCoordinator.cs

```diff
@@ -142,16 +143,23 @@ public void SuspendRequested(object sender, RubberduckStatusSuspendParserEventAr
                 var originalStatus = State.Status;
                 if (!e.AllowedRunStates.Contains(originalStatus))
                 {
-                    e.Result = SuspensionResult.IncompatibleState;
+                    e.Result = SuspensionOutcome.IncompatibleState;
                     return;
                 }
                 _parserStateManager.SetStatusAndFireStateChanged(e.Requestor, ParserState.Busy,
                     CancellationToken.None);
                 e.BusyAction.Invoke();
             }
-            catch
+            catch (OperationCanceledException ex)
             {
-                e.Result = SuspensionResult.UnexpectedError;
+                e.Result = SuspensionOutcome.Canceled;
+                e.EncounteredException = ex;
+                throw;
+            }
+            catch (Exception ex)
+            {
+                e.Result = SuspensionOutcome.UnexpectedError;
```

Comment:
As noted earlier in my previous comment regarding the enum, the `UnexpectedError` would be returned even if the exception originated from other place besides the `e.BusyAction.Invoke()` on the line 151. Does it make sense to separate the exception thrown in other places from the one that we may get as a result of `e.BusyAction.Invoke()`?